Random failure of the `test_kbd_interrupt_in_wait_with_limited_concurrency` unit test

The following failures happens randomly from time to time. I don't know if it actually corresponds to a bug in the code or is it a problem of this particular unit test.

```
_____________ test_kbd_interrupt_in_wait_with_limited_concurrency ______________
async_runner = (reframe.frontend.executors.Runner(_max_retries=0, _policy=reframe.frontend.executors.policies.AsynchronousExecutionPo...cs.TestStats(...)@0x7fdbd38f2908)@0x7fdbd38f2c18, <unittests.test_policies._TaskEventMonitor object at 0x7fdbd38f2208>)
make_cases = <function make_cases.<locals>._make_cases at 0x7fdbd34edc80>
make_async_exec_ctx = <function make_async_exec_ctx.<locals>._make_async_exec_ctx at 0x7fdbd34f46a8>
    def test_kbd_interrupt_in_wait_with_limited_concurrency(
            async_runner, make_cases, make_async_exec_ctx):
        # The general idea for this test is to allow enough time for all the
        # four checks to be submitted and at the same time we need the
        # KeyboardInterruptCheck to finish first (the corresponding wait should
        # trigger the failure), so as to make the framework kill the remaining
        # three.
        ctx = make_async_exec_ctx(2)
        next(ctx)
    
        runner, _ = async_runner
        with pytest.raises(KeyboardInterrupt):
            runner.runall(make_cases([
                KeyboardInterruptCheck(), SleepCheck(10),
>               SleepCheck(10), SleepCheck(10)
            ]))
E           Failed: DID NOT RAISE <class 'KeyboardInterrupt'>
unittests/test_policies.py:525: Failed
=========================== short test summary info ============================
FAILED unittests/test_policies.py::test_kbd_interrupt_in_wait_with_limited_concurrency
================== 1 failed, 619 passed, 88 skipped in 38.85s ==================
```

--- a/unittests/fixtures.py
+++ b/unittests/fixtures.py
@@ -24,6 +24,9 @@ TEST_MODULES = os.path.join(
     os.path.dirname(os.path.realpath(__file__)), 'modules'
 )
 
+# Builtin configuration
+BUILTIN_CONFIG_FILE = 'reframe/core/settings.py'
+
 # Unit tests site configuration
 TEST_CONFIG_FILE = 'unittests/resources/settings.py'
 

--- a/unittests/test_argparser.py
+++ b/unittests/test_argparser.py
@@ -123,55 +123,58 @@ def extended_parser():
 
 
 def test_option_precedence(extended_parser):
-    with rt.temp_environment(variables={
-            'RFM_TIMESTAMP': '%F',
-            'RFM_NON_DEFAULT_CRAYPE': 'yes',
-            'RFM_MODULES_PRELOAD': 'a,b,c',
-            'RFM_CHECK_SEARCH_PATH': 'x:y:z'
+    with rt.temp_runtime(fixtures.BUILTIN_CONFIG_FILE):
+        with rt.temp_environment(variables={
+                'RFM_TIMESTAMP': '%F',
+                'RFM_NON_DEFAULT_CRAYPE': 'yes',
+                'RFM_MODULES_PRELOAD': 'a,b,c',
+                'RFM_CHECK_SEARCH_PATH': 'x:y:z'
 
-    }):
-        options = extended_parser.parse_args(
-            ['--timestamp=%FT%T', '--nocolor']
-        )
-        assert options.recursive is None
-        assert options.timestamp == '%FT%T'
-        assert options.non_default_craype is True
-        assert options.config_file is None
-        assert options.prefix is None
-        assert options.stagedir == '/foo'
-        assert options.module == ['a', 'b', 'c']
-        assert options.check_path == ['x', 'y', 'z']
-        assert options.colorize is False
+        }):
+            options = extended_parser.parse_args(
+                ['--timestamp=%FT%T', '--nocolor']
+            )
+            assert options.recursive is None
+            assert options.timestamp == '%FT%T'
+            assert options.non_default_craype is True
+            assert options.config_file is None
+            assert options.prefix is None
+            assert options.stagedir == '/foo'
+            assert options.module == ['a', 'b', 'c']
+            assert options.check_path == ['x', 'y', 'z']
+            assert options.colorize is False
 
 
 def test_option_with_config(extended_parser):
-    with rt.temp_environment(variables={
-            'RFM_TIMESTAMP': '%F',
-            'RFM_NON_DEFAULT_CRAYPE': 'yes',
-            'RFM_MODULES_PRELOAD': 'a,b,c',
-            'RFM_KEEP_STAGE_FILES': 'no'
-    }):
-        site_config = rt.runtime().site_config
-        options = extended_parser.parse_args(
-            ['--timestamp=%FT%T', '--nocolor']
-        )
-        options.update_config(site_config)
-        assert site_config.get('general/0/check_search_recursive') is False
-        assert site_config.get('general/0/timestamp_dirs') == '%FT%T'
-        assert site_config.get('general/0/non_default_craype') is True
-        assert site_config.get('systems/0/prefix') == '.'
-        assert site_config.get('general/0/colorize') is False
-        assert site_config.get('general/0/keep_stage_files') is False
+    with rt.temp_runtime(fixtures.BUILTIN_CONFIG_FILE):
+        with rt.temp_environment(variables={
+                'RFM_TIMESTAMP': '%F',
+                'RFM_NON_DEFAULT_CRAYPE': 'yes',
+                'RFM_MODULES_PRELOAD': 'a,b,c',
+                'RFM_KEEP_STAGE_FILES': 'no'
+        }):
+            site_config = rt.runtime().site_config
+            options = extended_parser.parse_args(
+                ['--timestamp=%FT%T', '--nocolor']
+            )
+            options.update_config(site_config)
+            assert site_config.get('general/0/check_search_recursive') is False
+            assert site_config.get('general/0/timestamp_dirs') == '%FT%T'
+            assert site_config.get('general/0/non_default_craype') is True
+            assert site_config.get('systems/0/prefix') == '.'
+            assert site_config.get('general/0/colorize') is False
+            assert site_config.get('general/0/keep_stage_files') is False
 
-        # Defaults specified in parser override those in configuration file
-        assert site_config.get('systems/0/stagedir') == '/foo'
+            # Defaults specified in parser override those in configuration file
+            assert site_config.get('systems/0/stagedir') == '/foo'
 
 
 def test_option_envvar_conversion_error(extended_parser):
-    with rt.temp_environment(variables={
-            'RFM_NON_DEFAULT_CRAYPE': 'foo',
-    }):
-        site_config = rt.runtime().site_config
-        options = extended_parser.parse_args(['--nocolor'])
-        errors = options.update_config(site_config)
-        assert len(errors) == 1
+    with rt.temp_runtime(fixtures.BUILTIN_CONFIG_FILE):
+        with rt.temp_environment(variables={
+                'RFM_NON_DEFAULT_CRAYPE': 'foo',
+        }):
+            site_config = rt.runtime().site_config
+            options = extended_parser.parse_args(['--nocolor'])
+            errors = options.update_config(site_config)
+            assert len(errors) == 1

